src/components/Attachments/AttachmentCarousel/Pager/index.tsx

```diff
@@ -1,75 +1,51 @@
-import PropTypes from 'prop-types';
 import React, {useEffect, useImperativeHandle, useMemo, useRef, useState} from 'react';
 import {View} from 'react-native';
+import type {NativeViewGestureHandlerProps} from 'react-native-gesture-handler';
 import {createNativeWrapper} from 'react-native-gesture-handler';
+import type {PagerViewProps} from 'react-native-pager-view';
 import PagerView from 'react-native-pager-view';
-import Animated, {runOnJS, useAnimatedProps, useAnimatedReaction, useEvent, useHandler, useSharedValue} from 'react-native-reanimated';
-import _ from 'underscore';
-import refPropTypes from '@components/refPropTypes';
+import type {AnimatedProps} from 'react-native-reanimated';
+import Animated, {runOnJS, useAnimatedProps, useAnimatedReaction, useSharedValue} from 'react-native-reanimated';
 import useThemeStyles from '@hooks/useThemeStyles';
 import AttachmentCarouselPagerContext from './AttachmentCarouselPagerContext';
+import usePageScrollHandler from './usePageScrollHandler';
 
-const AnimatedPagerView = Animated.createAnimatedComponent(createNativeWrapper(PagerView));
-
-function usePageScrollHandler(handlers, dependencies) {
-    const {context, doDependenciesDiffer} = useHandler(handlers, dependencies);
-    const subscribeForEvents = ['onPageScroll'];
-
-    return useEvent(
-        (event) => {
-            'worklet';
+type PagerViewPropsObject = {
+    [K in keyof PagerViewProps]: PagerViewProps[K];
+};
 
-            const {onPageScroll} = handlers;
-            if (onPageScroll && event.eventName.endsWith('onPageScroll')) {
-                onPageScroll(event, context);
-            }
-        },
-        subscribeForEvents,
-        doDependenciesDiffer,
-    );
-}
+type AnimatedNativeWrapperComponent<P extends Record<string, unknown>, C> = React.ForwardRefExoticComponent<
+    React.PropsWithoutRef<AnimatedProps<P> & NativeViewGestureHandlerProps> & React.RefAttributes<C>
+>;
 
-const noopWorklet = () => {
-    'worklet';
+const AnimatedPagerView = Animated.createAnimatedComponent(createNativeWrapper(PagerView)) as AnimatedNativeWrapperComponent<PagerViewPropsObject, PagerView>;
```

Comment:
Because the `createNativeWrapper` function ignores the generically inferred component type:

**`node_modules/react-native-gesture-handler/lib/typescript/handlers/createNativeWrapper.d.ts`**

```ts
export default function createNativeWrapper<P>(Component: React.ComponentType<P>, config?: Readonly<NativeViewGestureHandlerProps>): React.ForwardRefExoticComponent<React.PropsWithoutRef<P & NativeViewGestureHandlerProps> & React.RefAttributes<React.ComponentType<any>>>;
```

We could instead create a patch for the `react-native-gesture-handler` and try to fix this upstream..